### PR TITLE
[1LP][RFR] Fix tests

### PR DESCRIPTION
**Pull request by valaparthvi**
## Purpose or Intent

- __Fixing__ 
    1. Fix `test_canned_corresponds.py` tests by adding 2 new fixtures `setup_provider_temp_appliance` and `setup_provider_temp_appliance_modscope` to make sure provider is added on the right appliance while using temp appliances in the test.
    2. Fix `test_default_filters_reset` and `test_reset_report_menus` failing because of updates with wt.pf FlashMessages
    3. Block `test_vm_retirement_rest.py` with BZ(1805119)
    4. Fix `test_customization_paginator` by fixing the `AddDialogView::is_displayed`. Also add same fix to `EditDialogView::is_displayed`.
    5. Fix `test_vm_volume_free_space_less_than_20_percent` by restoring the deleted template used by the test.
    6. Fix `TestProvidersRESTAPI` by waiting for the data before testing further.

### PRT Run
{{ pytest: cfme/tests/intelligence/reports/test_canned_corresponds.py cfme/tests/configure/test_default_filters.py::test_default_filters_reset cfme/tests/infrastructure/test_providers.py::test_infra_discovery_screen cfme/tests/infrastructure/test_vm_retirement_rest.py cfme/tests/intelligence/test_download_report.py cfme/tests/webui/test_general_ui.py::test_provider_documentation cfme/tests/intelligence/reports/test_reports.py::test_vm_volume_free_space_less_than_20_percent cfme/tests/cloud/test_providers.py::test_display_network_topology cfme/tests/intelligence/reports/test_menus.py::test_reset_report_menus cfme/tests/cloud/test_providers.py::TestProvidersRESTAPI cfme/tests/automate/test_customization_paginator.py --long-running }}

### CODE REVIEW

**Comment by dajoRH** [comment]
I detected some fixture changes in commit b60d03064d0a743cc22fa29dd32b385cb68b9dba

 The *global* fixture **`setup_provider_temp_appliance`** was changed, but I didn't find where it's used.
 The *global* fixture **`setup_provider_temp_appliance_modscope`** was changed, but I didn't find where it's used.
The *local* fixture **`setup_vm`** is used in the following files:
- **cfme/tests/intelligence/reports/test_reports.py**

Please, consider creating a PRT run to make sure your fixture changes do not break existing usage :smiley:

**Comment by dajoRH** [comment]
I detected some fixture changes in commit 674717ca581367c16d3a6f4c7c238ecfd6d22613

 The *global* fixture **`setup_provider_temp_appliance`** was changed, but I didn't find where it's used.
 The *global* fixture **`setup_provider_temp_appliance_modscope`** was changed, but I didn't find where it's used.
The *local* fixture **`setup_vm`** is used in the following files:
- **cfme/tests/intelligence/reports/test_reports.py**

Please, consider creating a PRT run to make sure your fixture changes do not break existing usage :smiley:

**Comment by dajoRH** [comment]
I detected some fixture changes in commit 8478d27a7b7b736b0186f779084f6d1577b41814

 The *global* fixture **`setup_provider_temp_appliance`** was changed, but I didn't find where it's used.
 The *global* fixture **`setup_provider_temp_appliance_modscope`** was changed, but I didn't find where it's used.
The *local* fixture **`setup_vm`** is used in the following files:
- **cfme/tests/intelligence/reports/test_reports.py**

Please, consider creating a PRT run to make sure your fixture changes do not break existing usage :smiley:

**Comment by dajoRH** [comment]
Would you mind rebasing this Pull Request against latest master, please? :trollface:
*CFME QE Bot*

**Comment by dajoRH** [comment]
I detected some fixture changes in commit 88882723fa7cbffdc11ddf71a294fb2671159fb8

 The *global* fixture **`setup_provider_temp_appliance`** was changed, but I didn't find where it's used.
 The *global* fixture **`setup_provider_temp_appliance_modscope`** was changed, but I didn't find where it's used.
The *local* fixture **`setup_vm`** is used in the following files:
- **cfme/tests/intelligence/reports/test_reports.py**

Please, consider creating a PRT run to make sure your fixture changes do not break existing usage :smiley:

**Comment by dajoRH** [comment]
Lint report for commit 88882723fa7cbffdc11ddf71a294fb2671159fb8:

`cfme/automate/dialogs/__init__.py`:
- :red_circle: Line 1:1: **F401** *'widgetastic.utils.Version' imported but unused*
- :red_circle: Line 9:1: **F401** *'cfme.utils.version.VersionPicker' imported but unused*

Please, rectify these issues :smirk: .


**No commit flaws detected**.

*CFME QE Bot*

**Comment by dajoRH** [comment]
I detected some fixture changes in commit d001bd20f6ac05687f59b38f4851029e407bb280

 The *global* fixture **`setup_provider_temp_appliance`** was changed, but I didn't find where it's used.
 The *global* fixture **`setup_provider_temp_appliance_modscope`** was changed, but I didn't find where it's used.
The *local* fixture **`setup_vm`** is used in the following files:
- **cfme/tests/intelligence/reports/test_reports.py**

Please, consider creating a PRT run to make sure your fixture changes do not break existing usage :smiley:

File: cfme/tests/intelligence/reports/test_reports.py [hunk 1]
```diff
@@ -181,7 +181,7 @@ def setup_vm(configure_fleecing, appliance, provider):
     vm = appliance.collections.infra_vms.instantiate(
         name=random_vm_name(context="report", max_length=20),
         provider=provider,
-        template_name="env-rhel7-20-percent-full-disk-tpl",
+        template_name="env-rhel7-20-percent-full-disk-pvala-tpl",
```
Comment (digitronik) [inline cfme/tests/intelligence/reports/test_reports.py]: I think this should come from data yaml than hardcodding. We can include this under template session for provider. I'm marking this optional as just modifying but please raise separate PR for this.

File: cfme/fixtures/provider.py [hunk 2]
```diff
@@ -104,6 +104,8 @@ def enable_provider_regions(provider):
 def _setup_provider_verbose(request, provider, appliance=None):
     if appliance is None:
         appliance = store.current_appliance
+    else:
+        provider.appliance = appliance
```
Comment (digitronik) [inline cfme/fixtures/provider.py]: can you please add comment for this?

File: cfme/tests/intelligence/reports/test_reports.py (hunk 1 above)
Comment (valaparthvi) [reply cfme/tests/intelligence/reports/test_reports.py]: This template won't be used by any other test case unless you want to provision a VM which has less than 20% disk space, which is why I directly used the template name even though it's weird to look at it. But if you insist, we can move it to data yaml.

**Comment by dajoRH** [comment]
I detected some fixture changes in commit f5d065a62e1646242fe481b13ce25c502b546d71

 The *global* fixture **`setup_provider_temp_appliance`** was changed, but I didn't find where it's used.
 The *global* fixture **`setup_provider_temp_appliance_modscope`** was changed, but I didn't find where it's used.
The *local* fixture **`setup_vm`** is used in the following files:
- **cfme/tests/intelligence/reports/test_reports.py**

Please, consider creating a PRT run to make sure your fixture changes do not break existing usage :smiley:

File: cfme/tests/intelligence/reports/test_reports.py (hunk 1 above)
Comment (digitronik) [reply cfme/tests/intelligence/reports/test_reports.py]: yup better to take it from yaml... It will help in template maintenance as well.

**Pull request merged**